articles/100_ament.md

```diff
@@ -0,0 +1,176 @@
+
+ament consists of several separate parts:
+
+* [ament_package](https://github.com/ament/ament_package) is a Python package providing an API to parse the files containing the meta information.
+* [ament_cmake](https://github.com/ament/ament_cmake) contains a set of packages which provide CMake based functionality.
+  The features are split across several packages in order to ensure that they are cleanly separated from each other.
+  The main feature of the CMake packages is to automatically generate a CMake config file to pass information to downstream packages.
+* [ament_lint](https://github.com/ament/ament_lint) contains a set of packages which perform linting tasks.
+  For as set of linters a ROS 2 specific configuration is provided which can be used via the command line as well as from within CMake (commonly as a CTest).
+* [ament_tools](https://github.com/ament/ament_tools) is a Python package which provide command line tools to build, test and install packages.
+  It is similar to [catkin_tools](https://github.com/catkin/catkin_tools) and builds each package in a workspace in topological order.
+  Each different build system is integrated through extension points.
+  While it currently does not build packages in parallel that will be added in the future to speed up the build process.
+* [ament_index](https://github.com/ament/ament_index) contains a set of packages which provide API to access the [ament resource index](https://github.com/ament/ament_cmake/blob/master/ament_cmake_core/doc/resource_index.md).
+  Currently it only contains a Python implementation but other languages like C++ should follow.
+
+
+### Integrate arbitrary build systems
+
+Each package can utilize a different build system to perform the steps of configuring, building and installing.
+Currently supported are CMake and Python but support for others (e.g. autotools, plain Makefiles) will likely be added in the future.
+For each of them the native steps are being applied separately by the ament tools.
+
+
+### Environment creation
+
+Commonly the install space should follow the [Filesystem Hierarchy Standard (FHS)](http://www.pathname.com/fhs/).
+But that is just a recommendation and not enforced by ament.
+
+Depending on where the packages are being installed it might be necessary to setup the environment in order to find all resources.
+E.g. the location of installed executables should be on the `PATH`, installed Python code should be on the `PYTHONPATH` etc.
+
+Therefore each package can provide a shell script to setup the environment to match its needs.
+The shells scripts in the install space, which are generated by the ament tools, will then only iterate over the scripts from each package to setup the environment.
+
+
+### Symlinked install
+
+It is very important to make the development cycle of changing code, building and installing it and then run it to confirm the changes as efficient as possible.
```

Comment:
It is very important to _maximize the efficiency of_ the development cycle of changing code, building and installing it and then run it to confirm the changes.